sotopia/install/cli.py

```diff
@@ -0,0 +1,203 @@
+                author={{Xuhui Zhou and Hao Zhu and Leena Mathur and Ruohong Zhang and Haofei Yu and Zhengyang Qi and Louis-Philippe Morency and Yonatan Bisk and Daniel Fried and Graham Neubig and Maarten Sap}},
+                booktitle={{The Twelfth International Conference on Learning Representations}},
+                year={{2024}},
+                url={{https://openreview.net/forum?id=mM7VurbA4r}}
+            }}
+            """)
+
+        Path("redis-data").mkdir(exist_ok=True)
+        try:
+            subprocess.run(f"curl -L {url} -o redis-data/dump.rdb", shell=True)
+            console.log("Database downloaded successfully.")
+        except subprocess.CalledProcessError:
+            console.log("Database download failed. Please check the URL and try again.")
+    else:
+        Path("redis-data").mkdir(exist_ok=True)
+        console.log("Starting redis with an empty database.")
+
+    if use_docker:
+        try:
+            subprocess.run(
+                "docker run --name redis-stack -p 6379:6379 -p 8001:8001 -v ($pwd)/redis-data/:/data/ redis/redis-stack:latest",
+                shell=True,
+            )
+            console.log("Redis started successfully.")
+        except subprocess.CalledProcessError:
+            console.log("Redis start failed. Please check the logs and try again.")
+    else:
+        if system == "Darwin":
+            try:
+                subprocess.run("brew tap redis-stack/redis-stack", shell=True)
+                subprocess.run("brew install redis-stack", shell=True)
+                if load_database:
+                    subprocess.run(
+                        "mv redis-data/dump.rdb /opt/homebrew/var/db/redis-stack/dump.rdb",
+                        shell=True,
+                    )
+                subprocess.run("redis-stack-server --daemonize yes", shell=True)
+                console.log("Redis started successfully.")
+            except subprocess.CalledProcessError:
+                console.log("Redis start failed. Please check the logs and try again.")
+        elif system == "Linux":
+            try:
+                subprocess.run(
+                    "curl -fsSL https://packages.redis.io/redis-stack/redis-stack-server-7.2.0-v10.focal.x86_64.tar.gz -o redis-stack-server.tar.gz",
+                    shell=True,
+                )
+                subprocess.run("tar -xvzf redis-stack-server.tar.gz", shell=True)
+                if load_database:
+                    subprocess.run(
+                        "mv redis-data/dump.rdb ./redis-stack-server-7.2.0-v10/var/db/redis-stack/dump.rdb",
+                        shell=True,
+                    )
+                subprocess.run(
+                    "./redis-stack-server-7.2.0-v10/bin/redis-stack-server --daemonize yes",
+                    shell=True,
+                )
+            except subprocess.CalledProcessError:
+                console.log("Redis start failed. Please check the logs and try again.")
+
+
```

Comment:
In the end, we should tell people what to do to connect with the codebase

Reply:
The connection is handled by sotopia, so they don't need to do that themselves.

Reply:
Also, it would be good to tell people how to shut down the redis server and how to restart the redis server

Reply:
> The connection is handled by sotopia, so they don't need to do that themselves.

I am a bit confused, so they don't need to put conda env var anymore? This is kinda problematic IMO since people have less control over which database they wanna use in cases if they need to switch

Reply:
> Also, it would be good to tell people how to shut down the redis server and how to restart the redis server
That's not what sotopia-install script could handle. Please open an issue on documentionation.

Reply:
> I am a bit confused, so they don't need to put conda env var anymore? This is kinda problematic IMO since people have less control over which database they wanna use in cases if they need to switch

We could add the environment variables so that they don't need to set the variables manually. They would always have control over everything. They can change it or refuse to add the environment variables when prompted.

Reply:
Yeah, please do that!

Reply:
> > Also, it would be good to tell people how to shut down the redis server and how to restart the redis server
> 
> That's not what sotopia-install script could handle. Please open an issue on documentation.

At least we need to tell people what to do here? (`console.log`)?

Reply:
The direction to shutdown and restart are added.